### PR TITLE
Enhance UI with CustomTextBox and improve build process

### PR DESCRIPTION
**Pull request by ItsSimko**
## Summary by Sourcery

Enhance the UI by integrating a new CustomTextBox widget, improving text handling and user interaction. Refactor the settings window for better organization and add tooltips for improved user guidance. Update the build process to include a no-console option and modify the CI workflow to handle alpha releases.

New Features:
- Introduce a CustomTextBox widget with a built-in copy button for enhanced text handling in the UI.

Enhancements:
- Improve the UI by integrating the CustomTextBox widget, replacing standard text boxes for better user interaction.
- Enhance the settings window with a more organized layout, separating LLM and Whisper settings into distinct frames.
- Add tooltips to various UI elements for better user guidance.

Build:
- Update the build process to include a no-console option in the PyInstaller command.

CI:
- Modify the CI workflow to handle alpha releases by checking tag patterns and setting the prerelease flag accordingly.

### CODE REVIEW

**Comment by ItsSimko** [comment]
@sourcery-ai review

File: src/FreeScribe.client/client.py [hunk 1]
```diff
@@ -245,40 +248,48 @@ def save_audio():
...
 
+DEFUALT_BUTTON_COLOUR = None
+
 def toggle_recording():
-    global is_recording, recording_thread
+    global is_recording, recording_thread, DEFUALT_BUTTON_COLOUR
     if not is_recording:
-        user_input.configure(state='normal')
-        user_input.delete("1.0", tk.END)
+        user_input.scrolled_text.configure(state='normal')
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:256]: Legacy code leaving as is

File: src/FreeScribe.client/client.py [hunk 2]
```diff
@@ -316,11 +327,11 @@ def send_audio_to_server():
         print("Using Local Whisper for transcription.")
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:327]: legacy code leaving as is

File: src/FreeScribe.client/UI/SettingsWindowUI.py [hunk 3]
```diff
@@ -126,71 +135,146 @@ def add_scrollbar_to_frame(self, frame):
...
+            self.models_drop_down.grid_remove()
+        elif self.models_drop_down.get() != "Custom" and self.custom_model_entry.grid_info() != {}:
+            # Hide the custom model entry
+            self.models_drop_down.grid(row=self.custom_model_entry.grid_info()['row'], 
+                        column=1, padx=0, pady=5, sticky="w")
+            self.custom_model_entry.grid_remove()
+
+    def get_selected_model(self):
+        """Returns the selected model, either from dropdown or custom entry"""
+        if self.models_drop_down.get() == "Custom" or self.models_drop_down.get() == "Select a Model":
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/UI/SettingsWindowUI.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/26/commits/f2c9f19de0a4c0c2a3f97edc2ae7385c55bb03ec

File: src/FreeScribe.client/ContainerManager.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ def __init__(self):
         try:
             self.client = docker.from_env()
         except docker.errors.DockerException as e:
-            print(f"An error occurred while initializing the Docker client, removing status bar: {e}")
+            client = None
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/ContainerManager.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/26/commits/cebde6332579799198afd14084aa8ff03617d1dc

File: src/FreeScribe.client/UI/SettingsWindowUI.py [hunk 5]
```diff
@@ -128,69 +130,102 @@ def add_scrollbar_to_frame(self, frame):
...
     def create_basic_settings(self):
         """
-        Creates the basic settings UI elements.
-
-        This method creates and places UI elements for basic settings such as
-        OpenAI API Key, and SSL settings.
+        Creates the basic settings UI elements in a two-column layout.
+        Settings alternate between left and right columns for even distribution.
         """
+        # Create left and right frames for the two columns
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/UI/SettingsWindowUI.py:161]: I will keep it as is.

File: src/FreeScribe.client/client.py [hunk 6]
```diff
@@ -127,15 +129,16 @@ def threaded_send_audio_to_server():
     thread.start()
 
 
-
+DEFUALT_PAUSE_BUTTON_COLOUR = None
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/FreeScribe.client/client.py:132]: Yes this should be fixed

**Review by yingbull — CHANGES_REQUESTED** [review]

File: src/FreeScribe.client/client.py [hunk 7]
```diff
@@ -595,21 +625,32 @@ def toggle_view():
         upload_button.grid()
         response_display.grid()
         timestamp_listbox.grid()
-        copy_user_input_button.grid()
-        copy_response_display_button.grid()
+        # copy_user_input_button.grid()
+        # copy_response_display_button.grid()
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/26/commits/062df80e4538c9512767470e156f6c586f40c9b8

**Comment by ItsSimko** [comment]
@sourcery-ai

**Comment by ItsSimko** [comment]
@sourcery-ai review

**Review by yingbull — APPROVED** [review]

**Pull request merged**